### PR TITLE
`core:sync` Add recursive mutex functions to `sync` procedure groups

### PR DESCRIPTION
**Pull request by deadwanderer**

### CODE REVIEW

File: core/sync/sync_util.odin [hunk 1]
```diff
@@ -6,9 +6,10 @@ Example:
 		...
 	}
 */
-guard :: proc{
+guard :: proc {
```
Comment (gingerBill) [inline core/sync/sync_util.odin:9]: Why did you add random spaces?

Comment (deadwanderer) [reply core/sync/sync_util.odin:9]: That was my VSCode auto-formatting, sorry.
